optimizely/helpers/condition.py

```diff
@@ -70,13 +91,29 @@ def exact_evaluator(self, index):
         - if the condition value or user attribute value has an invalid type.
         - if there is a mismatch between the user attribute type and the condition value type.
     """
+    condition_name = self.condition_data[index][0]
     condition_value = self.condition_data[index][1]
-    user_value = self.attributes.get(self.condition_data[index][0])
+    user_value = self.attributes.get(condition_name)
+
+    if not self.is_value_valid_for_exact_conditions(condition_value):
+      return None
+
+    if not self.is_value_valid_for_exact_conditions(user_value):
+      self.logger.warning(logs.UNEXPECTED_TYPE.format(
+          self._get_condition_log(index),
+          condition_name,
+          user_value
+      ))
+      return None
 
-    if not self.is_value_valid_for_exact_conditions(condition_value) or \
-       not self.is_value_valid_for_exact_conditions(user_value) or \
-       not validator.are_values_same_type(condition_value, user_value):
-        return None
+    if not validator.are_values_same_type(condition_value, user_value):
+      self.logger.debug(logs.MISMATCH_TYPE.format(
```

Comment:
Added this log to distinguish exactly what is inapplicable and what is type mismatch for exact evaluator. 
Not in docs. 
For exact, user value {}, [], () are inapplicable but integer 5 is a mismatch if condition value is a string